test/helpers/check-merge-safety.test.ts

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2021 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Mocktokit } from '../types';
+import { context } from '@actions/github';
+import { checkMergeSafety } from '../../src/helpers/check-merge-safety';
+import { octokit } from '../../src/octokit';
+import { setCommitStatus } from '../../src/helpers/set-commit-status';
+import { paginateAllOpenPullRequests } from '../../src/utils/paginate-open-pull-requests';
+
+const branchName = 'some-branch-name';
+
+jest.mock('../../src/utils/paginate-open-pull-requests');
+jest.mock('../../src/helpers/set-commit-status');
+jest.mock('@actions/core');
+jest.mock('@actions/github', () => ({
+  context: { repo: { repo: 'repo', owner: 'owner' }, issue: { number: 123 } },
+  getOctokit: jest.fn(() => ({
+    rest: {
+      repos: {
+        compareCommitsWithBasehead: jest.fn()
+      },
+      pulls: {
+        get: jest.fn(() => ({
+          data: {
+            base: { repo: { default_branch: 'main', owner: { login: 'owner' } } },
+            head: { ref: branchName, user: { login: 'username' } }
+          }
+        }))
+      }
+    }
+  }))
+}));
+
+describe('checkMergeSafety', () => {
+  it('should throw error when branch is out of date on the provided project path', async () => {
+    (octokit.repos.compareCommitsWithBasehead as unknown as Mocktokit).mockImplementation(async ({ basehead }) => {
+      const changedFiles =
+        basehead === 'some-branch-name...main'
+          ? ['packages/package-1/src/file1.ts', 'packages/package-2/src/file.ts']
+          : ['README.md', 'packages/package-1/src/file2.ts'];
+      return {
+        data: {
+          files: changedFiles.map(file => ({ filename: file }))
+        }
+      };
+    });
+    await expect(
+      checkMergeSafety({
+        paths: 'packages/package-1',
```

Comment:
in this example, why would `paths` be `packages/package-1` and not `packages/package-1,packages/package-2`?

Reply:
just because it's unnecessary for the test to complete i guess? just trying to understand the usage of `paths`

Reply:
So this was just a very simple test case. In the next test case you can see that I pass package-1 and package-2.

I said this in the README, but `paths` defines file paths to all of a repo's co-dependent projects which could be affected by a PR

Reply:
ok we can talk offline about what that means for our use cases